src/boot/efi/boot.c

```diff
@@ -1741,13 +1741,9 @@ static void config_title_generate(Config *config) {
 
         /* set title */
         for (UINTN i = 0; i < config->entry_count; i++) {
-                CHAR16 *title;
-
                 FreePool(config->entries[i]->title_show);
-                title = config->entries[i]->title;
-                if (!title)
-                        title = config->entries[i]->id;
-                config->entries[i]->title_show = StrDuplicate(title);
+                config->entries[i]->title_show = StrDuplicate(
```

Comment:
oom check?

Reply:
boot.c is currently pretty rough territory regarding OOM handling: it didn't use to do any OOM checking at all. New code adds it, and this PR adds it to two functions too, but with @medhefgo I recently discussed that for sd-boot the better option is probably to add aborting allocators (or freezing allocators), since we are pretty much the sole owner of the machine at the point we run and must work for the machine to continue booting. Thus, trying to handle OOM gracefully is not going to help much, since if you are the only component around and you fail then the whole system fails anyway: there isn't benefit in keeping failure local, if your locality is the only one around.

So. For now I think we probably should add OOM checks when we touch/write code where it's easy and obvious. But I'd shy away from bigger refactorings adding OOM checks given that sooner or later we want to have aborting allocators in sd-boot anyway, in which case they'd go away again anyway.

Hence yeah, coming back to this specific case: adding OOM checks is OK here, but not necessary I would say, given this would propagate into a slew of other changes (i.e. the function would have to return errors, and all its callers would have too) and I am not sure it's really necessary to pay the price for that now.

Reply:
ok, makes sense